lib/md059.mjs

```diff
@@ -0,0 +1,41 @@
+// @ts-check
+
+import { addErrorContext, stripAndDowncaseText } from "../helpers/helpers.cjs";
+import { validLink } from "../helpers/micromark-helpers.cjs";
+import { getReferenceLinkImageData, filterByTypesCached } from "./cache.mjs";
+
+const defaultBannedNames = [
+  "read more",
+  "learn more",
+  "more",
+  "here",
+  "click here",
+  "link"
+];
+
+/** @type {import("markdownlint").Rule} */
+export default {
+  "names": [ "MD059", "no-generic-link-name" ],
+  "description": "Links should have a descriptive name",
+  "tags": [ "links", "accessibility" ],
+  "parser": "micromark",
+  "function": function MD059(params, onError) {
+    const { definitions } = getReferenceLinkImageData();
+
+    // Allow consumer to define their own banned names list.
+
+    const bannedNames = params.config.banned_names_override || defaultBannedNames;
```

Comment:
I think calling it `banned_names` is even better.

Reply:
Updating this to `banned_names` will give me an ESLint error - `Identifier 'banned_names' is not in camel case.`

Reply:
I'm referring to the config parameter - instead of `banned_names_override`, just `banned_names`.